FlowCrypt/Controllers/Inbox/InboxViewController.swift

```diff
@@ -243,11 +258,16 @@ extension InboxViewController {
 // MARK: - Functionality
 extension InboxViewController {
     private func getSearchQuery() -> String? {
-        guard searchedExpression.isNotEmpty else { return nil }
+        let showOnlyPgp = UserDefaults.standard.bool(forKey: "SHOW_PGP_ONLY_FLAG")
+        let pgpPattern = """
+            ("-----BEGIN PGP MESSAGE-----" AND "-----END PGP MESSAGE-----") OR \
+            ("-----BEGIN PGP SIGNED MESSAGE-----") OR \
+            filename:({asc pgp gpg key})
+        """
+        let baseQuery = showOnlyPgp ? "\(pgpPattern) AND \(searchedExpression)" : searchedExpression
```

Comment:
currently `onlyPgp` mode works correctly only for the initial message loading.
when you scroll to the bottom of messages list and next page is loaded - it loads plain messages too:
<img width="68" alt="Screenshot 2024-11-08 at 11 10 22" src="https://github.com/user-attachments/assets/1221fea8-0898-4045-a1dd-d8c1e34575ab">

Reply:
Yep. Nice catch :+1: